src/main/python/mlsearch/api_requester.py

```diff
@@ -232,18 +245,23 @@ def _fetch_youtube(self, y_next_page_token=None) -> [Protocol]:
         if not self._config.YOUTUBE_FIX_KEYWORD.strip() in user_query:
             user_query = input_query + self._config.YOUTUBE_QUERY_FILTER
 
+        sampled_dev_key = None
+        if len(self._config.YOUTUBE_DEVELOPER_KEY) > 0:
+            sampled_dev_key = random.choice(self._config.YOUTUBE_DEVELOPER_KEY)
         youtube = googleapiclient.discovery.build(
             self._config.YOUTUBE_SERVICE_NAME,
             self._config.YOUTUBE_API_VERSION,
-            developerKey=self._config.YOUTUBE_DEVELOPER_KEY,
+            developerKey=sampled_dev_key,
```

Comment:
I'm not really sure about this one, but with the algorithm above, there's a possibility that `sampled_dev_key` could be `None`. I checked the API (https://googleapis.github.io/google-api-python-client/docs/epy/googleapiclient.discovery-module.html#build) and it looks like it accepted `None`?
Maybe we can remove the condition in L249 since it's the configuration that always available there. We can throw an exception if the config is not available.

Reply:
I confirmed the behavior and it is not possible to use `None`. When the client side did not provide the `YouTube Dev Key` then, it will use the keys provided from Lambda. If `None` is applied, then the `Http Error` Exception will be executed.  I am not sure what do you meant by the condition at L249?

Reply:
I confirmed the behavior and it is not possible to use `None`. When the client side did not provide the `YouTube Dev Key` then, it will use the keys provided from Lambda. If `None` is applied, then the `Http Error` Exception will be executed. I am not sure what do you meant by the condition at L249?

Reply:
Ah. Sorry for not being clear.  I'm pointing to this condition. `if len(self._config.YOUTUBE_DEVELOPER_KEY)`
It's just a matter of style though. 
at L248 you defined `sample_dev_key` as `None` and check if the keys are available `self._config.YOUTUBE_DEVELOPER_KEY`. If there are no keys available, then it'll be `None`, and error will be raised.
I do think it's configuration error, but I think it'll be great if we can raise an error before building the API client.
how do you think?